### ISSUE TITLE
Mac OSX 3.7 CI fail

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#1527]
## 🐛 Bug

We are getting a consistent Mac OSX 3.7 CI fail

See https://github.com/napari/napari/runs/951608936

looks like it might be pooch/ scikit-image related

```
Collecting monotonic>=0.1
  Using cached monotonic-1.5-py2.py3-none-any.whl (5.3 kB)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /Users/anka/conda/lib/python3.7/site-packages (from requests->pooch->-r requirements/test.txt (line 11)) (1.25.9)
Requirement already satisfied: idna<3,>=2.5 in /Users/anka/conda/lib/python3.7/site-packages (from requests->pooch->-r requirements/test.txt (line 11)) (2.10)
Requirement already satisfied: certifi>=2017.4.17 in /Users/anka/conda/lib/python3.7/site-packages (from requests->pooch->-r requirements/test.txt (line 11)) (2020.6.20)
Requirement already satisfied: chardet<4,>=3.0.2 in /Users/anka/conda/lib/python3.7/site-packages (from requests->pooch->-r requirements/test.txt (line 11)) (3.0.4)
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    scikit-image from https://files.pythonhosted.org/packages/9f/21/62a5e3b6b5a84d182c181d9787f9b0210d44ff2e2d6305a08a6664253266/scikit_image-0.17.2-cp37-cp37m-macosx_10_13_x86_64.whl#sha256=87ca5168c6fc36b7a298a1db2d185a8298f549854342020f282f747a4e4ddce9 (from -r requirements/test.txt (line 10)):
        Expected sha256 87ca5168c6fc36b7a298a1db2d185a8298f549854342020f282f747a4e4ddce9
             Got        2a227617e44a184943f657c0175ec7c279f596d4debc614f229b91fc310dfb7c
```

**Comment by sofroniewn** [comment]
Looks like CI just passed, so it might not happen every time, or it might no longer be an issue .... https://github.com/napari/napari/runs/956001037

**Comment by sofroniewn** [comment]
Ok it has failed again, https://github.com/napari/napari/runs/956641527, so looks like it might not be every time, but right now at 66%. It's also only Mac OSX Python 3.7 that fails, all the others always work

**Comment by tlambert03** [comment]
might be worth opening up a branch, enabling py3.7 on cirrus so it can be tested... and then maybe turn off the pip cache?

https://github.com/napari/napari/blob/2315443d9bd4fb6b8cc30967cf433d017f7c8ca2/.cirrus.yml#L61-L65

**Comment by sofroniewn** [comment]
Just noting that it looks now like we have many more CI fails due to a conda issue, see https://github.com/napari/napari/pull/1510/checks?check_run_id=968244108

```
wheel -> setuptools -> python_abi=3.8[build=*_cp38]
conda[version='>=4.8.4'] -> python_abi=3.8[build=*_cp38]
pip -> setuptools -> python_abi=3.8[build=*_cp38]
setuptools -> python_abi=3.8[build=*_cp38]
python_abi

Package tqdm conflicts for:
conda[version='>=4.8.4'] -> conda-package-handling[version='>=1.3.0'] -> tqdm
tqdm
conda-package-handling -> tqdm

Package requests conflicts for:
conda[version='>=4.8.4'] -> requests[version='>=2.18.4,<3']
requests

Package setuptools conflicts for:
setuptools
pip -> setuptools
wheel -> setuptools
conda[version='>=4.8.4'] -> setuptools[version='>=31.0.1']

Package ipaddress conflicts for:
cryptography -> ipaddress
requests -> urllib3[version='>=1.21.1,<1.26,!=1.25.0,!=1.25.1'] -> ipaddress
urllib3 -> ipaddress
pyopenssl -> cryptography[version='>=2.8'] -> ipaddress

Package archspec conflicts for:
conda[version='>=4.8.4'] -> archspec
archspec

Package brotlipy conflicts for:
requests -> urllib3[version='>=1.21.1,<1.26,!=1.25.0,!=1.25.1'] -> brotlipy[version='>=0.6.0']
urllib3 -> brotlipy[version='>=0.6.0']
brotlipy

Package pysocks conflicts for:
urllib3 -> pysocks[version='>=1.5.6,<2.0,!=1.5.7']
pysocks
requests -> urllib3[version='>=1.21.1,<1.26,!=1.25.0,!=1.25.1'] -> pysocks[version='>=1.5.6,<2.0,!=1.5.7']

Package yaml conflicts for:
ruamel_yaml -> yaml[version='>=0.1.7,<0.2.0a0|>=0.2.5,<0.3.0a0']
yaml
conda[version='>=4.8.4'] -> ruamel_yaml[version='>=0.11.14,<0.16'] -> yaml[version='>=0.1.7,<0.2.0a0|>=0.2.5,<0.3.0a0']

Package conda-package-handling conflicts for:
conda-package-handling
conda[version='>=4.8.4'] -> conda-package-handling[version='>=1.3.0']

Package chardet conflicts for:
requests -> chardet[version='>=3.0.2,<3.1.0|>=3.0.2,<4']
chardet
conda[version='>=4.8.4'] -> requests[version='>=2.18.4,<3'] -> chardet[version='>=3.0.2,<3.1.0|>=3.0.2,<4']
```

which will mask the above fail.

I think the conda issue will probably get fixed by the conda team in the next couple of days and we don't need to dig into, but it might make fixing this issue hard. See discussion here https://napari.zulipchat.com/#narrow/stream/212875-general/topic/CI.20errors/near/206502873 with @pwinston and @tlambert03

**Comment by sofroniewn** [comment]
Following #1541 we're back to only have the CI issue originally mentioned at the top of this issue.

**Comment by liu-ziyang** [comment]
can someone please rerun the failed macos ci task? I want to see if the cache is fixed without having to merge my PR on this

**Comment by tlambert03** [comment]
which one?  I just triggered this one https://cirrus-ci.com/task/5336673233928192

**Comment by liu-ziyang** [comment]
> which one? I just triggered this one https://cirrus-ci.com/task/5336673233928192
> 
> > <img alt="" width="16" height="16" src="https://avatars2.githubusercontent.com/in/3232?s=32&amp;u=d8bbf0d9c7c1a343bcb3a8e7405a93ed44521fd4&amp;v=4">
> >         Cirrus CI
> >       
> >     [Remove](#)
> >   
> >   
> >       [Task is in SCHEDULED status](https://cirrus-ci.com/task/5336673233928192)
> >   
> > 
> >   
> >     
> >       
> >         No instructions have finished!
> >       
> >       
> >        Show more
> >        Show less

https://github.com/napari/napari/runs/956641527
Thanks in advance!

**Comment by liu-ziyang** [comment]
It looks like https://cirrus-ci.com/task/5336673233928192 made it through, but I am not sure if because me successfully invalidated the cache, I will keep an eye on this issue

**Comment by liu-ziyang** [comment]
we need to merge https://github.com/napari/napari/pull/1548 in order for cirrus to clear the cache

**Comment by sofroniewn** [comment]
Closed by #1553
